Runtime/Core/SelectionPickerSettings.cs

```diff
@@ -1,70 +1,93 @@
         public ProBuilderMesh mesh;
-        public int vertex;
-        public Edge edge;
-        public Face face;
+        public List<int> vertexes;
+        public List<Edge> edges;
+        public List<Face> faces;
 
         public SceneSelection(GameObject gameObject = null)
         {
             this.gameObject = gameObject;
+            vertexes = new List<int>();
+            edges = new List<Edge>();
+            faces = new List<Face>();
         }
 
-        public SceneSelection(ProBuilderMesh mesh, int vertex) : this(mesh != null ? mesh.gameObject : null)
+        public SceneSelection(ProBuilderMesh mesh, List<int> vertexes) : this(mesh != null ? mesh.gameObject : null)
         {
             this.mesh = mesh;
-            this.vertex = vertex;
-            edge = Edge.Empty;
-            face = null;
+            this.vertexes = vertexes;
+            edges = new List<Edge>();
+            faces = new List<Face>();
         }
 
-        public SceneSelection(ProBuilderMesh mesh, Edge edge) : this(mesh != null ? mesh.gameObject : null)
+        public SceneSelection(ProBuilderMesh mesh, List<Edge> edges) : this(mesh != null ? mesh.gameObject : null)
         {
             this.mesh = mesh;
-            vertex = -1;
-            this.edge = edge;
-            face = null;
+            vertexes = new List<int>();
+            this.edges = edges;
+            faces = new List<Face>();
         }
 
-        public SceneSelection(ProBuilderMesh mesh, Face face) : this(mesh != null ? mesh.gameObject : null)
+        public SceneSelection(ProBuilderMesh mesh, List<Face> faces) : this(mesh != null ? mesh.gameObject : null)
         {
             this.mesh = mesh;
-            vertex = -1;
-            edge = Edge.Empty;
-            this.face = face;
+            vertexes = new List<int>();
+            edges = new List<Edge>();
+            this.faces = faces;
+        }
+
+        public void SetSingleFace(Face face)
+        {
+            faces.Clear();
+            faces.Add(face);
+        }
+
+        public void SetSingleVertex(int vertex)
+        {
+            vertexes.Clear();
+            vertexes.Add(vertex);
+        }
+
+        public void SetSingleEdge(Edge edge)
+        {
+            edges.Clear();
+            edges.Add(edge);
         }
 
         public void Clear()
         {
             gameObject = null;
             mesh = null;
-            face = null;
-            edge = Edge.Empty;
-            vertex = -1;
+            faces.Clear();
+            edges.Clear();
+            vertexes.Clear();
         }
 
         public void CopyTo(SceneSelection dst)
         {
             dst.gameObject = gameObject;
             dst.mesh = mesh;
-            dst.face = face;
-            dst.edge = edge;
-            dst.vertex = vertex;
+            dst.faces = faces.ConvertAll(x => x);
+            dst.edges = edges.ConvertAll(x => x);
+            dst.vertexes = vertexes.ConvertAll(x => x);
```

Comment:
You might see a performance improvement here if instead of using linq you just cleared and appended to the existing lists. Ex,

```
dst.faces.Clear(); 
foreach(var x in src.faces) 
    dst.faces.Add(x);
```